redirect.go

```diff
@@ -0,0 +1,217 @@
+	return r.c.flashMessages[key]
+}
+
+// Get old input data.
+func (r *Redirect) OldInputs() map[string]string {
+	return r.c.oldInput
+}
+
+// Get old input data by key.
+func (r *Redirect) OldInput(key string) string {
+	return r.c.oldInput[key]
+}
+
+// Redirect to the URL derived from the specified path, with specified status.
+func (r *Redirect) To(location string) error {
+	r.c.setCanonical(HeaderLocation, location)
+	r.c.Status(r.status)
+
+	return nil
+}
+
+// Route redirects to the Route registered in the app with appropriate parameters.
+// If you want to send queries or params to route, you should use config parameter.
+func (r *Redirect) Route(name string, config ...RedirectConfig) error {
+	// Check config
+	cfg := RedirectConfig{}
+	if len(config) > 0 {
+		cfg = config[0]
+	}
+
+	// Get location from route name
+	location, err := r.c.getLocationFromRoute(r.c.App().GetRoute(name), cfg.Params)
+	if err != nil {
+		return err
+	}
+
+	// Flash messages
+	if len(r.messages) > 0 || len(r.oldInput) > 0 {
+		messageText := bytebufferpool.Get()
+		defer bytebufferpool.Put(messageText)
+
+		// flash messages
+		i := 1
+		for k, v := range r.messages {
+			if i != 1 {
+				_, _ = messageText.WriteString("k:")
+			}
+
+			_, _ = messageText.WriteString(k + ":" + v)
+			if len(r.messages) != i {
+				_, _ = messageText.WriteString(",")
+			}
+			i++
+		}
+
+		// old input data
+		i = 1
+		for k, v := range r.oldInput {
+			if len(r.messages) > 1 && i == 1 {
+				_, _ = messageText.WriteString(",k:")
+			} else if i != 1 {
+				_, _ = messageText.WriteString("k:")
+			}
+
+			_, _ = messageText.WriteString("old_input_data_" + k + ":" + v)
+			if len(r.oldInput) != i {
+				_, _ = messageText.WriteString(",")
+			}
+			i++
+		}
+
+		r.c.Cookie(&Cookie{
+			Name:        "fiber_flash",
+			Value:       r.c.app.getString(messageText.Bytes()),
+			SessionOnly: true,
+		})
+	}
+
+	// Check queries
+	if len(cfg.Queries) > 0 {
+		queryText := bytebufferpool.Get()
+		defer bytebufferpool.Put(queryText)
+
+		i := 1
+		for k, v := range cfg.Queries {
+			_, _ = queryText.WriteString(k + "=" + v)
+
+			if i != len(cfg.Queries) {
+				_, _ = queryText.WriteString("&")
+			}
+			i++
+		}
+
+		return r.To(location + "?" + r.c.app.getString(queryText.Bytes()))
+	}
+
+	return r.To(location)
+}
+
+// Redirect back to the URL to referer.
+// TODO: Should fallback be optional?
+func (r *Redirect) Back(fallback string) error {
+	location := r.c.Get(HeaderReferer)
+	if location == "" {
+		location = fallback
+	}
+	return r.To(location)
+}
+
+// setFlash is a method to get flash messages before removing them
+func (r *Redirect) setFlash() {
+	// parse flash messages
+	if r.c.Cookies("fiber_flash") != "" {
+		messages := strings.Split(r.c.Cookies("fiber_flash"), ",k:")
+		r.c.flashMessages = make(map[string]string)
+		r.c.oldInput = make(map[string]string)
+
+		for _, msg := range messages {
+			splitMsg := strings.Split(msg, ":")
```

Comment:
think here many allocations are produced, can we use the logic which we also use with the "Accept" or "ExtractIP" ?